modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -827,6 +790,61 @@ TensorMechanicsAction::actGatherActionParameters()
   }
 }
 
+void
+TensorMechanicsAction::actAddVariable()
+{
+  // Add displacement variables (optional)
+  if (_is_primary_action && _current_task == "add_variable")
```

Comment:
Is add variable only called a single time for the first tensor mechanics action?  Will that work for separate blocks.

Reply:
yes